pkg/wrangler/context.go

```diff
@@ -290,31 +297,18 @@ func (w *Context) WithAgent(userAgent string) *Context {
 	wContextCopy.API = wContextCopy.api.WithAgent(userAgent).V1()
 	wContextCopy.CRD = wContextCopy.crd.WithAgent(userAgent).V1()
 	wContextCopy.Plan = wContextCopy.plan.WithAgent(userAgent).V1()
-	if w.DeferredCAPIRegistration.CAPIInitialized {
+
+	if w.DeferredCAPIRegistration.CAPIInitialized() {
 		wContextCopy.CAPI = wContextCopy.capi.WithAgent(userAgent).V1beta1()
 	} else {
-		go wContextCopy.BackfillCAPI(&wContextCopy)
+		wContextCopy.DeferredCAPIRegistration.DeferFunc(&wContextCopy, func(clients *Context) {
```

Comment:
This seems to attempt to backfill `wContextCopy` from itself. I am not sure if that works.

Reply:
> but my understanding is that the clients parameter is just a pointer to `wContextCopy`.

Actually there is a situation where it will not be.

When the `Deferred*APIRegistration` is initialized then what you said is true.

However, when it is not initialized then the `clients` parameter is __not saved__ by the `DeferRegistration/Func`, and when the actual execution happens later the system uses the wrangler context used with `ManageDeferred*APIContext` instead. Which is likely `w` here.

This is also why I get `copy/clients == w` (instead of `copy/clients == wContextCopy` as expected) when my function is called.

To get around this I should explicitly write `wContextCopy.Ext = w.ext.WithAgent(userAgent).V1()` and ignore the `copy/clients` data as unreliable.

Reply:
Ah, good catch! We may or may not drop this call when https://github.com/rancher/rancher/pull/51390#discussion_r2274097397 is implemented, but for the near term I'll definitely update it